packages/flutter/lib/src/widgets/animated_switcher.dart

```diff
@@ -48,9 +48,11 @@ typedef Widget AnimatedSwitcherTransitionBuilder(Widget child, Animation<double>
 /// Signature for builders used to generate custom layouts for
 /// [AnimatedSwitcher].
 ///
-/// The function should return a widget which contains the given children, laid
-/// out as desired. It must not return null.
-typedef Widget AnimatedSwitcherLayoutBuilder(List<Widget> children);
+/// The builder should return a widget which contains the given children, laid
+/// out as desired. It must not return null. The list of previous children is
+/// sorted from oldest to newest. The builder should be able to handle an empty
+/// list of previousChildren, or a null currentChild.
```

Comment:
We should explicitly say that previousChildren does not contain currentChild.

Reply:
We should also say if it's safe to mutate the previousChildren list, since a natural thing to do would be to add currentChild to that list if it's not null.